### PR TITLE
Propose: add the KubeVela project as a integration

### PR DESCRIPTION
**Pull request by barnettZQG**
**What type of PR is this?**

/kind documentation

**What this PR does / why we need it**:

The KubeVela community focuses on the practice of application specification, and we agree with the Gateway API in terms of the application gateway rules, so we integrate the specification.

**Does this PR introduce a user-facing change?**:

NONE

### CODE REVIEW

**Comment by robscott** [comment]
Hey @barnettZQG, thanks for the PR! Trying to understand how this integration works, could it be used with any Gateway API implementation or is it really just tied to Traefik? So far the integrations we have listed can be used with any implementation of the API.

**Comment by barnettZQG** [comment]
@robscott thanks for your reply, we provide the user's APIs  (the trait in OAM, including `http-route`, `https-route`, `tcp-route`, `udp-route`) based on Gateway API. The traefik is only a provider, we will integrate other providers that support the Gateway API. For users, they only change the provider name, and no need to change other configurations, when they move the application to other environments.

**Comment by robscott** [comment]
Hey @barnettZQG, I brought this up in today's community meeting, some notes:

1. We haven't clearly defined what qualifies something to be listed as an integration yet
2. We want to limit it to integrations that support multiple implementations of the API

I'm going to work on a PR to resolve the first item. It sounds like you're going to add support for additional Gateway API implementations soon, maybe we can just leave this PR open until KubeVela is able to support more than Traefik?

**Comment by barnettZQG** [comment]
@robscott OK, we will do it.

**Comment by barnettZQG** [comment]
@robscott Hi, I am sorry to update it after long time. Currently, we make an addon based on the Gateway API and provide two implementation addons. Traefik and Istio Gateway. Based on the same user-side experience, we could integrate more implementations.

File: site-src/implementations.md [hunk 1]
```diff
@@ -245,3 +247,15 @@ cert-manager can generate TLS certificates for Gateway resources. This is config
 
 [cert-manager]:https://cert-manager.io/
 [cert-manager-docs]:https://cert-manager.io/docs/usage/gateway/
+
+### KubeVela
+
+[KubeVela][kubevela] is a modern software delivery platform that makes deploying and operating applications across today's hybrid, multi-cloud environments easier, faster and more reliable.
+
+KubeVela using the Gateway API provides the standard management ability regarding the application access rules. It currently publishes a [gateway-traits][addon] to provide the user API. There are two implementation addon, [Traefik][traefik] and [Istio gateway][istio]. You can refer to the [installation document][document] to install the KubeVela and enable the gateway-traits addon.
```
Comment (gyohuangxin) [inline site-src/implementations.md:255]: A small typo.
```suggestion
KubeVela using the Gateway API provides the standard management ability regarding the application access rules. It currently publishes a [gateway-traits][addon] to provide the user API. There are two implementation addons, [Traefik][traefik] and [Istio gateway][istio]. You can refer to the [installation document][document] to install the KubeVela and enable the gateway-traits addon.
```

**Comment by Xunzhuo** [comment]
/lgtm

Need to resolve some conflicts and a rebase.

**Comment by shaneutt** [comment]
> /hold after [kubevela/catalog#598](https://github.com/kubevela/catalog/issues/598) gets finished.

I see we're waiting on https://github.com/kubevela/catalog/issues/598 before this PR can move forward. Looking at that issue it appears there is a significant amount of work that needs to be completed. Any thoughts on maybe closing this PR for the moment, so it doesn't have to linger, and then opening a PR when this is ready?

**Comment by barnettZQG** [comment]
@shaneutt OK. I will open a new PR after https://github.com/kubevela/catalog/issues/598 is complete.

**Pull request closed without merging**